esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,373 @@
+        {
+            if (!filterNestedScopeTypes || nestedScopeTypes.bSearch(scopeType,funcCompareUnsigned)==NotFound)
+              continue;
+        }
+
+        if (filterByAttributeValue)
+        {
+            Owned<CAttributeFilter> tmpForAttribSearch = new CAttributeFilter(stat.getKind());
+            aindex_t pos = arrayAttributeFilter.bSearch(tmpForAttribSearch,funcCompareAttribFilter);
+            if (pos!=NotFound)
+            {
+                CAttributeFilter & attributeFilter = arrayAttributeFilter.item(pos);
+                unsigned __int64 graphRawValue = stat.getValue();
+                if (!attributeFilter.matchesValue(graphRawValue))
+                    continue;
+            }
+        }
+        if (returnSpecifyReturnAttribs)
+        {
+            unsigned sk = (unsigned) stat.getKind();
+            if (returnAttribList.bSearch(sk,funcCompareUnsigned)==NotFound)
+               continue;
+        }
+
+        unsigned __int64 timeStamp = stat.getTimestamp();
+        if (timeStamp<minVersion)
+            continue;
+        if (timeStamp>maxTimestamp)
+            maxTimestamp = timeStamp;
+
+        if (nestedDepth && !foundInNestedScopeList)
+        {
+            bool isNew;
+            const char * tscope = strdup(scope.str());
+            nestedScopeList.bAdd(tscope, funcCompareString, isNew); // binary adds only if it is new
+        }
+        Owned<IEspWUResponseAttribute> attrib = copyAttribs(stat, attribOptions);
+        attribs.append(*attrib.getClear());
+        ++statsAdded;
+    }
+    DBGLOG("Statistics processed %u, stats added %u", statsCount, statsAdded);
+    if (previousScope.length()>0 && attribs.ordinality()>0)
+    {
+        Owned<IEspWUScope> respScope = buildRespScope(previousScope.str(), previousScopeType.str(), previousId.str(), attribs);
+        respScopes.append(*respScope.getClear());
+        attribs.clear();
+    }
+}
+
+IEspWUScope * CWUDetails::buildRespScope(const char * scope,
+                                         const char * scopeType,
+                                         const char * id,
+                                         IArrayOf<IEspWUResponseAttribute> & attribs)
+{
+    Owned<IEspWUScope> wuscope = createWUScope("","");
+
+    if (scopeOptions.getIncludeScope())     wuscope->setScope(scope);
+    if (scopeOptions.getIncludeScopeType()) wuscope->setScopeType(scopeType);
+    if (scopeOptions.getIncludeId())        wuscope->setId(id);
+    //if (scopeOptions.getIncludeMatchedScopesInResults()
+
+    wuscope->setAttributes(attribs);
+
+    return wuscope.getClear();
+}
+
+IEspWUResponseAttribute * CWUDetails::copyAttribs(IConstWUStatistic & cur,
+                                                  IConstWUAttributeOptions & attribOptions)
+{
+    Owned<IEspWUResponseAttribute> attrib = createWUResponseAttribute("","");
+    SCMStringBuffer tmpstr;
+
+    if (attribOptions.getIncludeName())
+        attrib->setName(queryStatisticName(cur.getKind()));
```

Comment:
nasty, but you need to check that cur.getKind() is a known attribute - otherwise looking at a new workunit on an old esp will cause problems.  Probably need to check in the stats loop.  Could argue that the stats iterator should filter them.